api/v1alpha6/types.go

```diff
@@ -316,3 +321,14 @@ type APIServerLoadBalancer struct {
 	// AllowedCIDRs restrict access to all API-Server listeners to the given address CIDRs.
 	AllowedCIDRs []string `json:"allowedCidrs,omitempty"`
 }
+
+// ValueSpec represents a single value_spec key-value pair.
+type ValueSpec struct {
+	// Name is the name of the key-value pair.
+	// This is just for identifying the pair and will not be sent to the OpenStack API.
```

Comment:
This needs:

```go
  // +required
  // +kubebuilder:validation:Required
```

Although tbh even after reading [the documentation](https://book.kubebuilder.io/reference/markers/crd-validation.html) I don't know what the difference is between those 2, but they always seem to go together 🤷

Reply:
@JoelSpeed I don't suppose you happen to know 👆?

Reply:
This does not seem to make any difference. I tried `make generate` with and without and nothing in the CRDs change. All fields are already correctly marked as required. Maybe there is a default somewhere? I see a lot of `+optional` but could nto find a single occurrence of `+required` in the repo :thinking:

Reply:
There's no such things as `+required`, but `+kubebuilder:validation:Required` should be added for completeness. Yes, the default is to be required, but someone could, per API group, decide to change the default to optional at some point, in which case, having defined required fields as explicitly required, is safer (and also more obvious that you have actively thought about whether it should be required or optional)

Reply:
Thanks for the explanation! I added `+kubebuilder:validation:Required`.